### ISSUE TITLE
How do I specify a destination via --xcargs?

### ISSUE THREAD

**Issue by gobetti** [peripheryapp/periphery#246]
Hi! I'd like my periphery command to start like
```
periphery scan \
  --xcargs "-destination generic/platform=iOS"
```
but that yields `Error: Missing value for '--xcargs <xcargs>'`

after experimentation, I've found that the culprit of this error is in the first dash character, i.e. the error does not occur if I input `destination` instead of `-destination`

However, in doing that the xcodebuild process will complain later, I guess due to the missing dash:
```
xcodebuild: error: Unknown build action 'destination'.
```

Note: I seem to need this for my iOS project because Periphery 2.3.3 is selecting macOS by default (#100):
```
Reason: The run destination My Mac is not valid for Running the scheme 'myLib-iOS'.
```

**Comment by ileitch** [comment]
Thanks for reporting. Yes, the behavior with `-` is a bug. But also, you generally shouldn't need to explicitly set `-destination`, as `xcodebuild` by default should chose one that's compatible with your project. Possibly your project has a misconfiguration. Do all targets set the `SDKROOT`, `SUPPORTED_PLATFORMS` and `IPHONEOS_DEPLOYMENT_TARGET` options?

**Comment by ileitch** [comment]
Do any of your targets have `SUPPORTS_MACCATALYST =  YES`?

**Comment by ileitch** [comment]
This is now fixed in 2.4.0. Instead of using `--xargs`, simply pass the arguments following the `--` terminator, so `periphery scan --xcargs --foo` is now `periphery scan -- --foo`

**Comment by gobetti** [comment]
The final app target has `SUPPORTS_MACCATALYST` explicitly set to No, and so does the library that periphery/xcodebuild complained about "My Mac" run destination, but there were other libraries that had it as Yes by default (not explicitly set in the pbxproj). Forcing those to No didn't help, unfortunately.

In any case, thank you very much for your fix, it does resolve this issue. I had to specify it last and the mix of dashes and spaces doesn't read obvious, but the documentation may cover enough. The line I've appended to the full command was `-- -destination generic/platform=iOS`
